overlord/overlord.go

```diff
@@ -93,6 +101,36 @@ func loadState(backend state.Backend) (*state.State, error) {
 	return state.ReadState(backend, r)
 }
 
+// Run runs a loop to ensure the current state regularly through StateEngine Ensure().
+func (o *Overlord) Run() {
+	intv := ensureInterval
+	o.loopTomb.Go(func() error {
+		tim := time.NewTimer(intv)
+		for {
+			select {
+			case <-o.loopTomb.Dying():
+				return nil
+			case <-tim.C:
+			}
+			err := o.stateEng.Ensure()
+			if err != nil {
+				logger.Panicf("state engine ensure failed not recoverably: %v", err)
```

Comment:
If the state engine or one of its managers would like to blow up badly, they might do so by themselves. This feels more like a "normal" error that cannot be dealt with locally at that time. Logging makes sense, but we probably shouldn't panic here. At least not until we understand better the consequences of doing so.